models/pull.go

```diff
@@ -171,6 +171,92 @@ func (pr *PullRequest) loadProtectedBranch(e Engine) (err error) {
 	return
 }
 
+func (pr *PullRequest) requiredApprovals() int64 {
+	if pr.ProtectedBranch == nil {
+		if err := pr.loadProtectedBranch(x); err != nil {
+			log.Error("Error loading ProtectedBranch", err)
+		}
+	}
+	if pr.ProtectedBranch != nil {
+		return pr.ProtectedBranch.RequiredApprovals
+	}
+	return 0
+}
+
+// RequiresApproval returns true if this pull request requires approval, otherwise returns false
+func (pr *PullRequest) RequiresApproval() bool {
+	return pr.requiredApprovals() > 0
+}
+
+// GetReviewLabel returns the localization label for the review of this pull request
+func (pr *PullRequest) GetReviewLabel() string {
+	if pr.RequiresApproval() {
+		if pr.GetRejectedReviewsCount() > 0 {
+			return "repo.pulls.review_rejected"
+		}
+		if pr.GetGrantedApprovalsCount() >= pr.ProtectedBranch.RequiredApprovals {
+			return "repo.pulls.review_approved"
+		}
+		return "repo.pulls.review_required"
+	}
+	return "repo.pulls.review_approved" // by default
```

Comment:
This function should return some kind of status and translation strings should be in template file